tests/unit/certGen.unitTest.ts

```diff
@@ -706,6 +707,151 @@ describe("cert-gen", () => {
+                            .then((payload: any) => {
+                                expect(payload).to.deep.equal(expectedResult);
+
+                            });
+                    });
+                });
+            });
+
+            const testResult2: any = JSON.parse(event.Records[1].body);
+            context("and certificate Data is generated", () => {
+                context("and test-result contains a Major Defect with Dangerous defect rectified", () => {
+                    it("should return Certificate Data with PRSDefects list in Fail Data", () => {
+                        const expectedResult: any = {
+                            TestNumber: "W01A00310",
+                            TestStationPNumber: "09-4129632",
+                            TestStationName: "Abshire-Kub",
+                            CurrentOdometer: {
+                                value: 12312,
+                                unit: "kilometres"
+                            },
+                            IssuersName: "CVS Dev1",
+                            DateOfTheTest: "26.02.2019",
+                            CountryOfRegistrationCode: "gb",
+                            VehicleEuClassification: "M1",
+                            RawVIN: "XMGDE02FS0H012345",
+                            RawVRM: "BQ91YHQ",
+                            EarliestDateOfTheNextTest: "26.12.2019",
+                            ExpiryDate: "25.02.2020",
+                            SeatBeltTested: "Yes",
+                            SeatBeltPreviousCheckDate: "26.02.2019",
+                            SeatBeltNumber: 2,
+                            DangerousDefects: undefined,
+                            MajorDefects: [
+                                "1.1.a A registration plate: missing. Front."
+                            ],
+                            MinorDefects: [
+                                "54.1.d.i Power steering: reservoir is below minimum level. Axles: 7. Outer Nearside."
+                            ],
+                            AdvisoryDefects: [
+                                "5.1 Compression Ignition Engines Statutory Smoke Meter Test: null Dasdasdccc"
+                            ],
+                            PRSDefects: [
+                                "54.1.a.ii Power steering: not working correctly and obviously affects steering control. Axles: 7. Inner Offside. Asdasd"
+                            ]
+                        };
+
+                        return certificateGenerationService.generateCertificateData(testResult2, "FAIL_DATA")
+                            .then((payload: any) => {
+                                expect(payload).to.deep.equal(expectedResult);
+
+                            });
+                    });
+                });
+            });
+
+            const testResult3: any = JSON.parse(event.Records[2].body);
```

Comment:
we can replace the any with inferred  type ... 
Json.Parse (stringified is a shallow copy and secondly also cannot handle circular objects)
note even ECMA6 props spread will do a shallow copy. 

Can be noted and enhanced later